File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/BatteryChargeReporter.java
package com.mapbox.services.android.navigation.v5.internal.navigation;

import java.util.Timer;
import java.util.TimerTask;

class BatteryChargeReporter {

  private static final int NO_DELAY = 0;
  private final Timer timer;
  private final TimerTask task;

  BatteryChargeReporter(Timer timer, TimerTask task) {
    this.timer = timer;
    this.task = task;
  }

  void scheduleAt(long periodInMilliseconds) {
    timer.scheduleAtFixedRate(task, NO_DELAY, periodInMilliseconds);
  }

  void stop() {
    timer.cancel();
  }
}

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/BatteryChargeReporter.kt
package com.mapbox.services.android.navigation.v5.internal.navigation

import java.util.Timer
import java.util.TimerTask

internal class BatteryChargeReporter(
    private val timer: Timer,
    private val task: TimerTask
) {

    companion object {
        private const val NO_DELAY = 0L
    }

    fun scheduleAt(periodInMilliseconds: Long) {
        timer.scheduleAtFixedRate(task,
            NO_DELAY, periodInMilliseconds)
    }

    fun stop() {
        timer.cancel()
    }
}
